### PR TITLE
feat: add option to start video playback muted

### LINKED ISSUE #778: Add option to start video playback muted
Currently any video played will play with sound. The user should have the option to play videos muted instead.
By default this option should be disabled.

#### Requirements
- Add an option in the media settings
    - "start video playback muted"
- When instantiating the `VideoPlayerNotifier`, set the volume to 0 if this option is enabled

##### Hints
- Options are saved in `media_preferences.dart`
- When creating the `VideoPlayerNotifier` in `video_player_provider.dart`, pass the option to the notifier

### PR DESCRIPTION
**Pull request by S-ecki**
## Goals
- Add settings option to mute videos by default
- Reflect selection on video player

## Notes
- It was the first time working with Riverpod for me, I hope I understood the concepts correctly. Feel free to suggest improvements, I am open to feedback!

closes #778

### CODE REVIEW

File: lib/components/widgets/video_player/video_player_provider.dart [hunk 1]
```diff
@@ -13,10 +13,12 @@ final videoPlayerProvider = StateNotifierProvider.autoDispose
     .family<VideoPlayerNotifier, VideoPlayerState, VideoPlayerArguments>(
   (ref, arguments) {
     final handler = ref.watch(videoPlayerHandlerProvider);
+    final mediaPreferences = ref.watch(mediaPreferencesProvider);
```
Comment (robertodoering) [inline lib/components/widgets/video_player/video_player_provider.dart:16]: Watching the mediaPrefences with `ref.watch` here before the `VideoPlayerNotifier` gets constructed will cause the video player to get re-created every time the mediaPreferences update - which isn't exactly what we want.

We want to check whether the video should start muted only once when we initialize the video. Changing the setting afterwards shouldn't affect the video player.

To do this you can pass down the `ref` argument to the `VideoPlayerNotifier` and use `_ref.read(mediaPreferencesProvider)` in `initialize()` to get the value.

You can check out the `downloadPathProvider` for reference. We are doing something similar there.

**Review by robertodoering — COMMENTED** [review]
Looking great overall, nice job!
I have only a small remark about how to get the value

File: lib/components/widgets/video_player/video_player_provider.dart (hunk 1 above)
Comment (S-ecki) [reply lib/components/widgets/video_player/video_player_provider.dart]: Hey. Thanks for the great feedback and explanation, I understand the problem!
I added a new commit to fix it :)

**Review by robertodoering — APPROVED** [review]
Perfect! Thank you :)

**Pull request merged**